mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -7038,78 +7600,120 @@ struct ThreadwiseCopyV2RewritePattern
     //   llvm::errs() << sliceLengths[i] << " ";
     // llvm::errs() << "\n";
 
+    // Compute high-level coordinate for dest memref.
```

Comment:
This is lowering of present value of multi-index of the point of origin? This needs to be done only once when constructing the "thread-copy" object. It's not needed when calling "thread-copy.Run()" every time.

The trick is to build "lower-of-difference" inside "thread_copy.move_slice_window()"

Reply:
This is an optimization we can consider in subsequent PRs.